microlith/scripts/collect-information.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Couchbase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file  except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the  License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Don't set euo pipefail - we want this script to be as resilient as possible
+set -x
+
+echo "Starting collect-information.sh..."
+
+tmpdir=${TEMPORARY_DIRECTORY:-$(mktemp -d)}
+exec &> >(tee -a "$tmpdir/collect-information.sh.log")
+
+cp /etc/couchbase-cluster-monitor-release.txt "$tmpdir"
+cp /etc/cmos-release.txt "$tmpdir"
+
+# Environment
+env > "$tmpdir/env.txt"
+
+# Copy over all logs
+mkdir -p "$tmpdir/logs"
+cp /logs/* "$tmpdir/logs/"
+
+# Do not copy /var/log/nginx/*, because they get mapped to stdout/stderr and cp will hang forever
+# shellcheck disable=SC2043
+for var_log in grafana; do
+  for f in /var/log/"$var_log"/*; do cp "$f" "$tmpdir/logs/$var_log.$(basename "$f")"; done
+done
+
+# Configuration
+mkdir -p "$tmpdir/config"
+
+while IFS= read -r dir; do
+  cp -r "$dir" "$tmpdir/config/$(basename "$dir")/"
+done <<EOF
+/etc/alertmanager/
+/etc/prometheus/
+/etc/grafana/
+/etc/jaeger/
+/etc/loki/
+/etc/nginx/
+EOF
+
+# Grab various overridden paths
+mkdir -p "$tmpdir/dynamic-config"
+for override_var in PROMETHEUS_CONFIG_FILE PROMETHEUS_CONFIG_TEMPLATE_FILE JAEGER_CONFIG_FILE LOKI_CONFIG_FILE ALERTMANAGER_CONFIG_FILE; do
+  cp "${!override_var}" "$tmpdir/dynamic-config/$(basename "${!override_var}")"
+done
+
+# Entry points
+mkdir -p "$tmpdir/entrypoints"
+cp /entrypoints/* "$tmpdir/entrypoints/"
+
+# Misc scripts
+mkdir -p "$tmpdir/scripts"
+cp -r /scripts "$tmpdir/scripts/"
+cp /run.sh "$tmpdir/scripts/"
+cp /collect-information.sh "$tmpdir/scripts/"
+
+# Grafana plugins
+mkdir -p "$tmpdir/grafana-plugins"
+for d in /var/lib/grafana/plugins/*; do
+  cp "$d/plugin.json" "$tmpdir/grafana-plugins/$(basename "$d").json"
+done
+
+# Prometheus stats snapshot
+snapshot_file_name=$(curl -X POST http://localhost:9090/prometheus/api/v1/admin/tsdb/snapshot | jq -r '.data.name')
```

Comment:
I'd say make this optional, enabled by default - it might be large/people don't want to send it

Reply:
Actually a separate script we could use just to snapshot would be nice too... for auto testing updates.

Reply:
Hmm, I like the idea of a separate script, but I'm not sure if we should make this optional, otherwise we get into the situation of "well we _could_ tell what was wrong with your dashboards if only we had the stats snapshot" - cbcollects are huge anyway :p.

We do need to be careful about file size though, you're right.